### ISSUE TITLE
zKillboard will work with you

### ISSUE THREAD

**Issue by cvweiss** [evekb/evedev-kb#75]
Hey guys, had someone tell me about the issues you had with some of the recent API changes. Honestly, had I known this github existed I would have notified you guys and probably saved many some headaches and troubles. Sorry about that.

I'll try to let you know about any changes I make to the API when as I do them, or even ahead of time if I'm planning changes. Also, if you find any issues that I might not be aware of yet, please feel free to create an Issue at zkillboard's repo: https://github.com/zKillboard/zKillboard

EDK was my first killboard, I'll always have a warm place in my heart for it <3

**Comment by Salvoxia** [comment]
Hi,

thank you for that, I really appreciate it!
I think in the meantime we have successfully adapted to the latest API changes :)

Best Regards,
Salvoxia

**Comment by cvweiss** [comment]
FYI, the following parameters now require an entity to be passed first:

* pastSeconds
* startTime
* endTime
* beforeKillID
* afterKillID
* year
* month
* finalblow-only

So, just be sure to pass a characterID, corporationID, etc, with their values _before_ passing any of the above listed.

**Comment by Salvoxia** [comment]
Thanks for the heads-up!

**Comment by cvweiss** [comment]
FYI!

https://www.reddit.com/r/evetech/comments/71jw8i/zkill_warning_zkillboard_will_be_exporting_esi/

**Comment by Salvoxia** [comment]
Once again, thanks for the heads-up! 👍

**Comment by cvweiss** [comment]
FYI because of abuse, the orderDirection parameter is no longer allowed (yes, seriously...)

I noticed that EDK uses this in the killmail fetch, I highly recommended removing the parameter.

**Comment by cvweiss** [comment]
also, seriously? 

    xxx.xxx.xxx.xxx - - [28/Jun/2018:15:20:02 +0000] "GET /api/corporation/98067874/api-only/startTime/198011302141/orderDirection/asc/ HTTP/1.1" 200 55 "-" "EDK HTTP Requester, http://www.evekb.org" 0.205 0.205 

1980?

**Comment by Salvoxia** [comment]
Hi,

thanks for the heads up.
Will the kills still be ordered in any way?

And sorry for the 1980, it's possible for the user to input a date. I'll add a a sanity-check to that.

Best Regards,
Salvoxia

**Comment by cvweiss** [comment]
Killmails are default ordered by kill id descending. And I should have known that was user data!

I'm considering adding a recent endpoint that will get around the problem of older killmails being added quite recently and not being picked up. For example, something like:

    https://zkillboard.com/alliance/99005770/recent/

Would return the 200 most recent killmails added for that alliance. It would have a one hour cache point, and if I added pagination I would limit that to 10 as its very rare for an entity to have >2k killmails in one hour.

There's also the other options of RedisQ and/or subscribing to the websocket.
https://github.com/zKillboard/RedisQ (no filtering, but no worries about disconnects)
https://github.com/zKillboard/zKillboard/wiki/Websocket (has filtering)

In short, I'm removing api functionality because people are being very creative in finding ways around the caches and causing extra load on the server. Not sure why when there is RedisQ (for years now) and Websocket (recent addition) for getting new killmails... but I'm doing what I have to do.

**Comment by jbs1** [comment]
as of yesterday zkills api will not return the full killmail anymore.
this mean zkb fetch needs to be rewritten

https://github.com/zKillboard/zKillboard/wiki/API-(Killmails)#-attention-

**Comment by Salvoxia** [comment]
Thank you very much for the heads-up, zKBFetch has been updated to work with these changes (refer to https://github.com/evekb/evedev-kb/commit/76736e5a1e12596f082a390f796d6dbbd2a0a070 ).
Actually, only the kill timestamp was used of the kill data, so the changes weren't that extensive.

**Comment by cvweiss** [comment]
damn, I apologize I forgot to notify you folks of the change... 
I'm glad it didn't break things too much for you!

**Comment by cvweiss** [comment]
Since this code base is likely dead now :( I'm going to go ahead and close my open issue.
